Assets/Fungus/Scripts/Commands/VariableCondition.cs

```diff
@@ -6,49 +6,147 @@
+                conditions = new List<conditionExpression>();
+            }
+
+            if (conditions.Count == 0)
+            {
+                conditions.Add(new conditionExpression());
+            }
+        } 
 
         protected override bool EvaluateCondition()
         {
-            if (anyVar.variable == null)
+            if (conditions == null || conditions.Count == 0)
             {
                 return false;
             }
 
-            bool condition = false;
-            anyVar.Compare(compareOperator, ref condition);
+            bool resultAny = false, resultAll = true;
+            foreach (conditionExpression condition in conditions) 
+            {
+                bool curResult = false;
+                if (condition.AnyVar == null) 
+                {
+                    resultAll &= curResult;
+                    resultAny |= curResult;
+                    continue;
+                }
+                condition.AnyVar.Compare(condition.CompareOperator, ref curResult);
+                resultAll &= curResult;
+                resultAny |= curResult;
+            }
+
+            if (anyOrAllConditions == AnyOrAllConditions.AnyOneTrue) return resultAny;
             
-            return condition;
+            return resultAll;
         }
 
         protected override bool HasNeededProperties()
         {
-            return (anyVar.variable != null);
+            if (conditions == null || conditions.Count == 0)
+            {
+                return false;
+            }
+
+            foreach (conditionExpression condition in conditions) 
+            {
+                if (condition.AnyVar == null || condition.AnyVar.variable == null)
+                {
+                    return false;
+                }
+                
+            }
+            return true;
         }
 
         #region Public members
 
         /// <summary>
         /// The type of comparison operation to be performed.
         /// </summary>
-        public virtual CompareOperator CompareOperator { get { return compareOperator; } }
+        public virtual CompareOperator CompareOperator { get { return conditions[0].CompareOperator; } }
+
+        public virtual List<conditionExpression> Conditions { get { return conditions; } }
 
         public override string GetSummary()
         {
-            if (anyVar.variable == null)
+            if (!this.HasNeededProperties())
             {
                 return "Error: No variable selected";
             }
 
-            string summary = anyVar.variable.Key + " ";
-            summary += VariableUtil.GetCompareOperatorDescription(compareOperator) + " ";
-            summary += anyVar.GetDataDescription();
+            string summary = "";
+            string connector = "";
+            if (anyOrAllConditions == AnyOrAllConditions.AnyOneTrue)
+            {
+                connector = " Or ";
+            }
+            else
+            {
+                connector = " And ";
```

Comment:
These don't read easily in the command summary preview, try all caps and maybe wrap in bold tags